### ISSUE TITLE
BUILDING_NODE_EXTENSION is not defined

### ISSUE THREAD

**Issue by shinichy** [cmake-js/cmake-js#37]
node-gyp defines BUILDING_NODE_EXTENSION in addon.gypi (https://github.com/nodejs/node-gyp/blob/9bfa0876b46764978492fffea074d2d7aa8954f9/addon.gypi). It's used in node.h below.
cmake-js should define it, too.

``` c++
#ifdef _WIN32
# ifndef BUILDING_NODE_EXTENSION
#   define NODE_EXTERN __declspec(dllexport)
# else
#   define NODE_EXTERN __declspec(dllimport)
# endif
#else
# define NODE_EXTERN /* nothing */
#endif

#ifdef BUILDING_NODE_EXTENSION
# undef BUILDING_V8_SHARED
# undef BUILDING_UV_SHARED
# define USING_V8_SHARED 1
# define USING_UV_SHARED 1
#endif
```

**Comment by RSATom** [comment]
Why just not use [add_definitions](https://cmake.org/cmake/help/v3.0/command/add_definitions.html) ?

**Comment by shinichy** [comment]
I'm wondering if cmake-js can do it automatically as node-gyp does it. I don't need to define it manually if I use node-gyp.

**Comment by unbornchikken** [comment]
Good point. Expect that will land on this weekend, because I'm a bit tight of schedule right now.

**Comment by unbornchikken** [comment]
Unfortunately it's not as easy as I thought in the first place. I'm thinking on a module defined CMakeLists.txt that includes all common stuff from there: https://github.com/cmake-js/cmake-js#general and that BUILDING_NODE_EXTENSION definition as well.

**Comment by shinichy** [comment]
I think that's fine. Thanks for working on it!

**Comment by d3x0r** [comment]
https://github.com/nodejs/node/issues/20758
